src/Native/Unix/System.Native/pal_termios.cpp

```diff
@@ -0,0 +1,360 @@
+}
+
+extern "C" int32_t SystemNative_TermiosGetDcd(int fd)
+{
+    return SystemNative_TermiosGetSignal(fd, TIOCM_CAR);
+}
+
+extern "C" int32_t SystemNative_TermiosGetCts(int fd)
+{
+    return SystemNative_TermiosGetSignal(fd, TIOCM_CTS);
+}
+
+extern "C" int32_t SystemNative_TermiosGetRts(int fd)
+{
+    return SystemNative_TermiosGetSignal(fd, TIOCM_RTS);
+}
+
+extern "C" int32_t SystemNative_TermiosGetDsr(int fd)
+{
+    return SystemNative_TermiosGetSignal(fd, TIOCM_DSR);
+}
+
+extern "C" int32_t SystemNative_TermiosGetDtr(int fd)
+{
+    return SystemNative_TermiosGetSignal(fd, TIOCM_DTR);
+}
+
+extern "C" int32_t SystemNative_TermiosSetDtr(int fd, bool set)
+{
+    int status = SystemNative_TermiosGetStatus(fd);
+    if (status >= 0)
+    {
+        if (set)
+        {
+            status |= TIOCM_DTR;
+        }
+        else
+        {
+            status &= ~TIOCM_DTR;
+        }
+        return ioctl(fd, TIOCMSET, &status);
+    }
+
+    return -1;
+}
+
+extern "C" int32_t SystemNative_TermiosSetRts(int fd, bool set)
+{
+    int status = SystemNative_TermiosGetStatus(fd);
+    if (status >= 0)
+    {
+        if (set)
+        {
+            status |= TIOCM_RTS;
+        }
+        else
+        {
+            status &= ~TIOCM_RTS;
+        }
+        return ioctl(fd, TIOCMSET, &status);
+    }
+
+    return -1;
+}
+
+static speed_t
+SystemNative_TermiosSpeed2rate(int speed)
+{
+    switch (speed)
+    {
+    case 230400:
+        return B230400;
+    case 115200:
+        return B115200;
+    case 57600:
+        return B57600;
+    case 38400:
+        return B38400;
+    case 19200:
+        return B19200;
+    case 9600:
+        return B9600;
+    case 4800:
+        return B4800;
+    case 2400:
+        return B2400;
+    case 1800:
+        return B1800;
+    case 1200:
+        return B1200;
+    case 600:
+        return B600;
+    case 300:
+        return B300;
+    case 200:
+        return B200;
+    case 150:
+        return B150;
+    case 134:
+        return B134;
+    case 110:
+        return B110;
+    case 75:
+        return B75;
+    case 50:
+        return B50;
+    }
+    return (static_cast<speed_t>(speed));
+}
+
+static int
+SystemNative_TermiosRate2speed(speed_t brate)
+{
+    switch (brate)
+    {
+    case B230400:
+        return 230400;
+    case 115200:
+        return 115200;
+    case B57600:
+        return 57600;
+    case B38400:
+        return 38400;
+    case B19200:
+        return 19200;
+    case B9600:
+        return 9600;
+    case B4800:
+        return 4800;
+    case B2400:
+        return 2400;
+    case B1800:
+        return 1800;
+    case B1200:
+        return 1200;
+    case B600:
+        return 600;
+    case B300:
+        return 300;
+    case B200:
+        return 200;
+    case B150:
+        return 150;
+    case B134:
+        return 134;
+    case B110:
+        return 110;
+    case B75:
+        return 75;
+    case B50:
+        return 50;
```

Comment:
Do these constant B values not always have the same numerical value as what their name suggests?

Reply:
They don't, they have completely different values. They are enum-like, with values going up by one upto B38400. Above that, there is a large jump in the numbering and then they go again by one.